src/paratii.js

```diff
@@ -80,6 +85,87 @@ class Paratii extends ParatiiCore {
   getAccount () {
     return this.eth.getAccount()
   }
+  /**
+   * Pings the provider to which the web3 is configured to connect to (see the set up in paratii.eth.js constructor)
+   * @return {Promise} that resolves in a boolean
+   */
+  async pingEth () {
+    return new Promise(resolve => {
+      this.eth.web3.eth.net.isListening()
+        .then(() => {
+          resolve(true)
+        })
+        .catch(e => {
+          resolve(false)
+        })
+    })
+  }
+  /**
+   * Checks if the IPFS local node is running
+   * @return {Promise} that resolves in a boolean
+   */
+  async checkIPFSState () {
+    return new Promise(resolve => {
+      this.ipfs.getIPFSInstance()
+        .then((ipfsInstance) => {
+          if (ipfsInstance.state.state() === 'running') {
+            resolve(true)
+          } else {
+            resolve(false)
+          }
+        })
+        .catch(e => {
+          resolve(false)
+        })
+    })
+  }
+  /**
+   * Requests a link to see if it's up (Easily adds a dozen seconds to check the status)
+   * @param {string} linkToCheck
+   * @return {Promise} that resolves in a boolean
+   */
+  async requestStatusCode (linkToCheck) {
+    return new Promise(resolve => {
+      request(linkToCheck, function (error, response) {
+        if (error) {
+          resolve(false)
+        } else if (response && response.statusCode === 200) { // We suppose for now that only 200 is the right status code (see: https://en.wikipedia.org/wiki/List_of_HTTP_status_codes)
+          resolve(true)
+        } else {
+          resolve(false)
+        }
+      })
+    })
+  }
+  /**
+   * Checks the bootstrap dns nodes
+   * @param {string} baseUrl url of the web socket server
+   * @param {Number} port the port at which the web socket is listening to
+   * @return {Promise} that resolves in a boolean
+   */
+  async checkBootstrapWebSocketDNS (baseUrl, port) {
+    return new Promise(resolve => {
+      var client = new net.Socket()
+      client.setTimeout(30000) // Arbitrary 30 secondes to be able to reach DNS server
```

Comment:
why such a high number?

Reply:
In one test it took more than 10s to reach the dns, so I set it to 30. Perhaps I should lower it to around 20? i'm unsure as to which is the reasonable duration for the bootstrap server to respond

Reply:
I dunno, but 10s seems outrageous.  @ya7ya , enlighten us?